k8s-install/canal-etcdless.yaml

```diff
@@ -0,0 +1,177 @@
+# This ConfigMap can be used to configure a self-hosted Canal installation.
+# See `canal.yaml` for an example of a Canal deployment which uses
+# the config in this ConfigMap.
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: canal-config 
+  namespace: kube-system
+data:
+  # The interface used by canal for host <-> host communication.
+  # If left blank, then the interface is chosing using the node's
+  # default route.
+  canal_iface: ""
+
+  # Whether or not to masquerade traffic to destinations not within
+  # the pod network.
+  masquerade: "true"
+
+  # The CNI network configuration to install on each node.
+  cni_network_config: |-
+    {
+        "name": "k8s-pod-network",
+        "type": "calico",
```

Comment:
The flannel CNI plugin, when running in etcd mode, is just used for passing IPAM information to Calico.

Since IPAM is done entirely through the k8s API `Node.PodCIDR` allocations when using the kubernetes datastore driver (kdd), we don't need that chaining - both Calico and flannel look to the API as their source of truth.